### ISSUE TITLE
AbstractInterpreter handles logical operators incorrectly

### ISSUE THREAD

**Issue by saelo** [googleprojectzero/fuzzilli#213]
The AbstractInterpreter currently assumes that the [result of binary logical operators is a .boolean](https://github.com/googleprojectzero/fuzzilli/blob/87e9142747dd76387ca9d0c8d02acf3a78c5196c/Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift#L430). While that probably seems intuitive, that's not how JavaScript works:

```JavaScript
let a = 0 || "foobar";
print(a);     // "foobar"
let b = "" && 42;
print(b);     // ""
```

The correct semantics are explained for example [here for `||`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR) and [here for `&&`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND), but in essence, the operators always yield the value of one of the two input expressions, so probably the resulting type should simply be the union of the two inputs.

**Comment by Dhananjayshukla1** [comment]
@saelo I am new to this Organisation, can you please assign this to me.

**Comment by Dhananjayshukla1** [comment]
@amarekano I am new to this Organisation, can you please assign this to me

**Comment by Dhananjayshukla1** [comment]
@saelo I want to work on this issue, can you assign this to me.

**Comment by saelo** [comment]
That's great, thanks! We usually don't assign issues, but happy to do so if you like!

**Comment by Dhananjayshukla1** [comment]
Sorry sir,
I forgot my github account so i was not able to reply your message and your valuable time.
let a = 0 || "foobar";
console.log(a);     // "foobar"
let b = "" && 42;
console.log(b);     // ""

**Comment by Dhananjayshukla1** [comment]
I'm new to open source contributions but I'm well versed with C++, HTML, CSS, Javascript, React, and Node.js.
could you please tell me how to get started.

**Comment by nfgrusk** [comment]
Looks like this issue was resolved a good while ago (see #295). Perhaps, it’s time to close it. 😅

**Comment by saelo** [comment]
Indeed, looks like that fixes the issue. Thanks for pointing that out!
